### DISCUSSION TITLE
Upgrade to 1.7.36 version from 1.7.24 fails with "missing configuration" error

### DISCUSSION THREAD

**Discussion by chenchuanliang** [oracle/opengrok#4077] [General]
When I upgrade opengrok from 1.7.29 to 1.7.36 version,Replaced the 1.7.36 opengrok tools and change the tomcat/webapps/source.war,check the configuration in web.xml.
But opengrok still remind me this error:
```
"The source root path has not been configured! Please configure your webapp.
Unable to determine source root path. Missing configuration?
java.io.FileNotFoundException: Unable to determine source root path. Missing configuration?
	at org.opengrok.web.PageConfig.checkSourceRootExistence(PageConfig.java:1738) "
```
Previously, I upgraded from 1.7.24 to 1.7.29，no this issue.
Do I need to change anything？
Thanks

**Comment by vladak** [comment]
So, what does `web.xml` actually contain in terms of the `CONFIGURATION` parameter ? Does the file exist ? What is inside ? Does the Tomcat report any other problems in the logs ?

**Reply by chenchuanliang** [reply]
Yes, this file exists. Before upgrading, version 1.7.29 can run normally. After upgrading to version 1.7.36, the tomcat configuration has not changed. Continue to use the 1.7.29 CONFIGURATION file, which contains the project index information.

**Reply by chenchuanliang** [reply]
> So, what does `web.xml` actually contain in terms of the `CONFIGURATION` parameter ? Does the file exist ? What is inside ? Does the Tomcat report any other problems in the logs ?

I check the tomcat logs found maybe failed to parse CONFIGURATION xml file .

**Reply by vladak** [reply]
yep:
```
26-Oct-2022 18:25:49.939 信息 [main] org.opengrok.web.WebappListener.contextInitialized Starting webapp with version 1.7.36 (88f08643f5fb02d6ac014979ece732bb2f21215b)
26-Oct-2022 18:25:49.939 信息 [main] org.opengrok.indexer.configuration.Configuration.read Reading configuration from /home/unicair/opengrok/etc/configuration.xml
26-Oct-2022 18:25:50.483 警告 [main] org.opengrok.web.WebappListener.contextInitialized Configuration error. Failed to read config file: 
	java.io.IOException: java.lang.NullPointerException: target should not be null
		at org.opengrok.indexer.configuration.Configuration.decodeObject(Configuration.java:1477)
		at org.opengrok.indexer.configuration.Configuration.read(Configuration.java:1445)
		at org.opengrok.indexer.configuration.RuntimeEnvironment.readConfiguration(RuntimeEnvironment.java:1491)
```
Either you need to figure out the culprit in the config file or let the indexer create a new one.

**Reply by chenchuanliang** [reply]
It has been solved. There is a problem with the config file.Currently indexing, an old file is used.
Thank you

**Comment by chenchuanliang** [comment]
[catalina.txt](https://github.com/oracle/opengrok/files/9868615/catalina.txt)
tomcat  log
